modules/reactor/include/meshgenerators/ReportingIDGeneratorBase.h

```diff
@@ -0,0 +1,68 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+#include "libmesh/elem.h"
+#include "libmesh/replicated_mesh.h"
+#include "libmesh/dof_object.h"
+
+/**
+ * A base class that contains common members for reporting ID generators.
+ */
+class ReportingIDGeneratorBase
```

Comment:
This should probably be put in utils; it isn't actually a mesh generator and should just be a set of static methods, i.e.:

```
namespace ReportingIDGeneratorUtils
{
getCellwiseIntegerIDs(const std::vector<std::unique_ptr<ReplicatedMesh>> & meshes,
                      const std::vector<std::vector<unsigned int>> & pattern,
                      const bool use_exclude_id,
                      const std::vector<bool> & exclude_ids);
}

```

etc

Reply:
Thank you for suggestion. I will update it accordingly.